itest/lnd_payment_test.go

```diff
@@ -16,6 +17,124 @@ import (
+	// Grab Alice and Bob's nodes for convenience.
+	alice, bob := ht.Alice, ht.Bob
+
+	// Create a list of commitment types we want to test.
+	commitTyes := []lnrpc.CommitmentType{
+		lnrpc.CommitmentType_LEGACY,
+		lnrpc.CommitmentType_ANCHORS,
+		lnrpc.CommitmentType_SIMPLE_TAPROOT,
+	}
+
+	// testSendPayment opens a channel between Alice and Bob using the
+	// specified params. It then sends a payment from Alice to Bob and
+	// asserts it being successful.
+	testSendPayment := func(ht *lntest.HarnessTest,
+		params lntest.OpenChannelParams) {
+
+		// Check that there are no payments before test.
+		chanPoint := ht.OpenChannel(alice, bob, params)
+
+		// Now that the channel is open, create an invoice for Bob
+		// which expects a payment of 1000 satoshis from Alice paid via
+		// a particular preimage.
+		const paymentAmt = 1000
+		preimage := ht.Random32Bytes()
+		invoice := &lnrpc.Invoice{
+			RPreimage: preimage,
+			Value:     paymentAmt,
+		}
+		invoiceResp := bob.RPC.AddInvoice(invoice)
+
+		// With the invoice for Bob added, send a payment towards Alice
+		// paying to the above generated invoice.
+		payReqs := []string{invoiceResp.PaymentRequest}
+		ht.CompletePaymentRequests(alice, payReqs)
+
+		p := ht.AssertNumPayments(alice, 1)[0]
+		path := p.Htlcs[len(p.Htlcs)-1].Route.Hops
+
+		// Ensure that the stored path shows a direct payment to Bob
+		// with no other nodes in-between.
+		require.Len(ht, path, 1, "wrong number of routes in path")
+		require.Equal(ht, bob.PubKeyStr, path[0].PubKey, "wrong pubkey")
+
+		// The payment amount should also match our previous payment
+		// directly.
+		require.EqualValues(ht, paymentAmt, p.ValueSat,
+			"incorrect sat amount")
+		require.EqualValues(ht, paymentAmt*1000, p.ValueMsat,
+			"incorrect msat amount")
+
+		// The payment hash (or r-hash) should have been stored
+		// correctly.
+		correctRHash := hex.EncodeToString(invoiceResp.RHash)
+		require.Equal(ht, correctRHash, p.PaymentHash, "incorrect hash")
+
+		// As we made a single-hop direct payment, there should have
+		// been no fee applied.
+		require.Zero(ht, p.FeeSat, "fee should be 0")
+		require.Zero(ht, p.FeeMsat, "fee should be 0")
+
+		// Now verify that the payment request returned by the rpc
+		// matches the invoice that we paid.
+		require.Equal(ht, invoiceResp.PaymentRequest, p.PaymentRequest,
+			"incorrect payreq")
+
+		// Delete all payments from Alice. DB should have no payments.
+		alice.RPC.DeleteAllPayments()
+		ht.AssertNumPayments(alice, 0)
+
+		// TODO(yy): remove the sleep once the following bug is fixed.
+		// When the invoice is reported settled, the commitment dance
+		// is not yet finished, which can cause an error when closing
+		// the channel, saying there's active HTLCs. We need to
+		// investigate this issue and reverse the order to, first
+		// finish the commitment dance, then report the invoice as
+		// settled.
+		time.Sleep(2 * time.Second)
+
+		// Close the channel.
```

Comment:
Should we note here that this implicitly makes sure the channel is active (as coop-close is only possible when the channel is fully operational and online).

Reply:
Done!